### PR TITLE
Esa 68: AUIW to load a video into the video window

### PR DESCRIPTION
**Pull request by aameijide**

### CODE REVIEW

File: LongoMatch.Core/Common/EventsBrokerLongomatch.cs [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+using VAS.Core.Common;
+using VAS.Core.Handlers;
+using VAS.Core.Interfaces;
+using VAS.Core.Interfaces.GUI;
+using VAS.Core.Store;
+using VAS.Core.Store.Playlists;
+
+namespace LongoMatch.Core.Common
+{
+	public class EventsBroker : VAS.Core.Common.EventsBroker
```
Comment (vguzmanp) [inline LongoMatch.Core/Common/EventsBrokerLongomatch.cs]: There's another EventsBroker in the `EventsBroker.cs` file. Should we merge this into the other? Or is this one just a mistake and we can delete it?

File: LongoMatch.DB/LongoMatch.DB.Net45.csproj [hunk 2]
```diff
@@ -31,25 +29,28 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="System" />
-    <Reference Include="ICSharpCode.SharpZipLib.Portable">
-      <HintPath>..\oneplay-longomatch\packages\SharpZipLib.Portable.0.86.0.0003\lib\portable-net45+netcore45+wp8+win8+wpa81+MonoTouch+MonoAndroid+Xamarin.iOS10\ICSharpCode.SharpZipLib.Portable.dll</HintPath>
-    </Reference>
     <Reference Include="Newtonsoft.Json">
       <HintPath>..\oneplay-longomatch\packages\Newtonsoft.Json.7.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
```
Comment (vguzmanp) [inline LongoMatch.DB/LongoMatch.DB.Net45.csproj]: This is fixed in the `migration` branch already

File: LongoMatch.Drawing/CanvasObjects/Timeline/TimeNodeObject.cs [hunk 3]
```diff
@@ -15,147 +15,22 @@
...
-using LongoMatch.Core.Common;
-using VAS.Core.Interfaces.Drawing;
 using LongoMatch.Core.Store;
 using VAS.Core.Common;
-using VAS.Core.Store.Drawables;
 using VAS.Core.Store;
-using VAS.Core;
+using VAS.Core.Store.Drawables;
+using VASDrawing = VAS.Drawing;
+using VASTimeline = VAS.Drawing.CanvasObjects.Timeline;
```
Comment (vguzmanp) [inline LongoMatch.Drawing/CanvasObjects/Timeline/TimeNodeObject.cs]: Do we need the alias?

File: LongoMatch.Drawing/Widgets/CamerasTimeline.cs [hunk 4]
```diff
@@ -18,14 +18,15 @@
...
-using LongoMatch.Core.Handlers;
-using LongoMatch.Drawing.CanvasObjects.Timeline;
 using VAS.Core.Common;
 using VAS.Core.Handlers;
 using VAS.Core.Interfaces.Drawing;
 using VAS.Core.Store;
 using VAS.Core.Store.Drawables;
+using VAS.Drawing;
+using VAS.Drawing.CanvasObjects.Timeline;
+using LMTimeline = LongoMatch.Drawing.CanvasObjects.Timeline;
```
Comment (vguzmanp) [inline LongoMatch.Drawing/Widgets/CamerasTimeline.cs]: better to use as few selectors as possible in the rest of the code: `using TimeNodeObject = LongoMatch.Drawing.CanvasObjects.Timeline.TimeNodeObject`

File: LongoMatch.Core/Handlers/Handlers.cs [hunk 5]
```diff
@@ -57,6 +56,7 @@ namespace LongoMatch.Core.Handlers
 	public delegate void PlaylistVisibiltyHandler (bool visible);
 	public delegate void AnalysisWidgetsVisibilityHandler (bool visible);
 	public delegate void AnalysisModeChangedHandler (VideoAnalysisMode mode);
-	public delegate void ShowTimelineMenuHandler (List<TimelineEventLongoMatch> plays,EventType cat,Time time);
-	public delegate void ShowTaggerMenuHandler (List<TimelineEventLongoMatch> plays);
+	public delegate void ShowTimelineMenuHandler (List<TimelineEvent> plays,EventType cat,Time time);
+	public delegate void ShowTaggerMenuHandler (List<TimelineEvent> plays);
```
Comment (vguzmanp) [inline LongoMatch.Core/Handlers/Handlers.cs]: Would be better to have `IEnumerable<>` here

File: LongoMatch.Drawing/Widgets/PositionTagger.cs [hunk 6]
```diff
@@ -179,7 +182,7 @@ protected override void ShowMenu (Point coords)
 		{
 			if (ShowMenuEvent != null) {
 				List<TimelineEventLongoMatch> plays = Selections.Select (p => (p.Drawable as PositionObject).Play).ToList ();
-				ShowMenuEvent (plays);
+				ShowMenuEvent (plays.Cast<TimelineEvent> ().ToList ());
```
Comment (vguzmanp) [inline LongoMatch.Drawing/Widgets/PositionTagger.cs]: `plays` should be of type `List<TimelineEvent>` (actually, better ÌEnumerable<TimelineEvent>`). This way we avoid the need to `.Cast`

File: LongoMatch.GUI.Multimedia/Gui/PlayerCapturerBin.cs [hunk 7]
```diff
@@ -17,120 +17,34 @@
 // 
 using LongoMatch.Core.Store;
 using VAS.Core.Common;
-using VAS.Core.Interfaces;
-using VAS.Core.Interfaces.GUI;
-using VAS.UI.Helpers;
+using VASUi = VAS.UI;
```
Comment (vguzmanp) [inline LongoMatch.GUI.Multimedia/Gui/PlayerCapturerBin.cs]: One `.` less is not much of an improvement 😃

File: LongoMatch.GUI/Gui/Component/CodingWidget.cs [hunk 8]
```diff
@@ -207,7 +207,7 @@ public void AddPlay (TimelineEventLongoMatch play)
 		public void DeletePlays (List<TimelineEventLongoMatch> plays)
 		{
 			if (projectType == ProjectType.FileProject) {
-				timeline.RemovePlays (plays);
+				timeline.RemovePlays (plays.Cast<TimelineEvent> ().ToList ());
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Component/CodingWidget.cs]: I think this should use `OfType`
`Cast` will throw an exception when any of the elements is not castable
`OfType` will return only the castable elements

File: LongoMatch.GUI/Gui/Component/SynchronizationWidget.cs [hunk 9]
```diff
@@ -44,7 +46,7 @@ public partial class SynchronizationWidget : Gtk.Bin
 		Timerule timerule;
 		Time duration, currentTime, nextCurrentTime;
 		ProjectLongoMatch project;
-		PeriodsMenu menu;
+		PeriodsMenuBase menu;
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Component/SynchronizationWidget.cs]: `Base`

File: LongoMatch.GUI/Gui/GUIToolkit.cs [hunk 10]
```diff
@@ -36,6 +36,7 @@
 using VAS.UI.Helpers;
 using Image = VAS.Core.Common.Image;
 
+
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/GUIToolkit.cs]: extra line

File: LongoMatch.GUI/Gui/Menu/PlaysMenu.cs [hunk 11]
```diff
@@ -227,19 +219,19 @@ void CreateMenu ()
 			Add (addPLN);
 			
 			render = new MenuItem ("");
-			render.Activated += (sender, e) => EmitRenderPlaylist (plays);
+			render.Activated += (sender, e) => EmitRenderPlaylist (plays.Cast<TimelineEventLongoMatch> ().ToList ());
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Menu/PlaysMenu.cs]: `OfType`

File: LongoMatch.GUI/Gui/Menu/PlaysMenu.cs [hunk 12]
```diff
@@ -227,19 +219,19 @@ void CreateMenu ()
...
 			
 			snapshot = new MenuItem (Catalog.GetString ("Export to PNG images"));
-			snapshot.Activated += (sender, e) => ((LMCommon.EventsBroker)Config.EventsBroker).EmitSnapshotSeries (plays [0]);
+			snapshot.Activated += (sender, e) => ((LMCommon.EventsBroker)Config.EventsBroker).EmitSnapshotSeries (plays [0] as TimelineEventLongoMatch);
 			Add (snapshot);
 
 			duplicate = new MenuItem ("");
 			duplicate.Activated += (sender, e) => Config.EventsBroker.EmitDuplicateEvent (plays.Cast<TimelineEvent> ().ToList ());
-			Add (duplicate);
+ 			Add (duplicate);
```
Comment (vguzmanp) [inline LongoMatch.GUI/Gui/Menu/PlaysMenu.cs]: Extra espace

File: LongoMatch.GUI/gtk-gui/objects.xml [hunk 13]
```diff
@@ -322,7 +318,7 @@
     </itemgroups>
     <signals />
   </object>
-  <object type="LongoMatch.Gui.PlayerCapturerBin" palette-category="LongoMatch" allow-children="false" base-type="Gtk.Bin">
+  <object type="LongoMatch.Gui.PlayerCapturerBin" palette-category="General" allow-children="false" base-type="Gtk.Bin">
```
Comment (vguzmanp) [inline LongoMatch.GUI/gtk-gui/objects.xml]: Why the change of category?

File: LongoMatch.Services/RenderingJobsManager.cs [hunk 14]
```diff
@@ -24,6 +24,7 @@
 using VAS.Core.Interfaces.Multimedia;
 using VAS.Core.Store;
 using VAS.Core.Store.Playlists;
+using VASDrawing = VAS.Drawing;
```
Comment (vguzmanp) [inline LongoMatch.Services/RenderingJobsManager.cs]: alias not needed

File: LongoMatch.Drawing/CanvasObjects/Timeline/TimeNodeObject.cs (hunk 3 above)
Comment (aameijide) [inline LongoMatch.Drawing/CanvasObjects/Timeline/TimeNodeObject.cs]: VASTimeline is only used to indicate class inheritance, so I think that we can write it like:
public class TimeNodeObject : VAS.Drawing.CanvasObjects.Timeline.TimeNodeObject

Also, I deleted VASDrawing alias because it is not useful, it only saves us to write a dot.

File: LongoMatch.GUI/Gui/Component/CodingWidget.cs (hunk 8 above)
Comment (aameijide) [inline LongoMatch.GUI/Gui/Component/CodingWidget.cs]: I think in this case it doesn't matter as all TimelineEventLongoMatch are TimelineEvent, so cast should not fail never. So if it fails, maybe it's better to throw an exception because its something that is not expected

File: LongoMatch.GUI/Gui/Component/SynchronizationWidget.cs (hunk 9 above)
Comment (aameijide) [inline LongoMatch.GUI/Gui/Component/SynchronizationWidget.cs]: Periodsmenu doesn't have sense in VAS, so PeriodsMenu is moved again to Longomatch

As a future work, we should move Period class to Longomatch (currently is marked with FIXME).

File: LongoMatch.Drawing/CanvasObjects/Timeline/TimeNodeObject.cs (hunk 3 above)
Comment (vguzmanp) [inline LongoMatch.Drawing/CanvasObjects/Timeline/TimeNodeObject.cs]: Why not add a `using VAS.Drawing.CanvasObjects.Timeline` ?

Comment (aameijide) [inline LongoMatch.Drawing/CanvasObjects/Timeline/TimeNodeObject.cs]: Even writing the using, as LongoMatch and VAS clases have the same name, Xamarin interpretates it like a circular dependency and cannot compile.

Like it's used only to indicate the inheritance, I consider it ok to leave it with all the namespace

**Comment by vguzmanp** [comment]
:+1:

**Comment by vguzmanp** [comment]
:+1:
